core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -393,7 +273,7 @@ def generate_vagrant_file(path, config, provider, cookbook_path)
   # otherwise - ERROR_RESULT or ARGUMENT_ERROR_RESULT.
   def generate(path, config, override, provider)
     # TODO: MariaDb Version Validator
-    checks_result = check_path(path, override) && check_nodes_names(config)
+    checks_result = AttributesGenerator.check_path(path, override, @ui) && AttributesGenerator.check_nodes_names(config, @ui)
```

Comment:
You call AttributeGenerator the same way in Vagrant and Terraform. It is clear that you should provide a high-level method that will do both checks.